packages/node/src/starknet/block.starknet.ts

```diff
@@ -86,8 +86,9 @@ export function filterTransactionsProcessor(
     if (filter.function) {
       const index = decodedCalls?.findIndex(
         (call) =>
-          hexEq(call.selector, filter.function!) ||
-          hexEq(call.selector, encodeSelectorToHex(filter.function!)),
+          !!filter.function &&
```

Comment:
Wont this always be true? Something seems off with the typescript checks. 

The same applies to all the other cases.